src/transformStateWithClones.js

```diff
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let correctState = Object.assign({}, state);
+
+  for (const action of actions) {
+    // let correctState = Object.assign({}, state);
+    // console.log(correctState);
+    switch (action['type']) {
+      case 'addProperties':
+        correctState = Object.assign(correctState, action['extraData']);
+
+        const cloneAdd = Object.assign({}, correctState);
+
+        result.push(cloneAdd);
```

Comment:
why do you make copy inside each case? you sould do it only once after all cases.
also you can use spread operator to make copy here